### PR TITLE
Refactor Travis config to use stages and Python 3.7

### PR DESCRIPTION
**Pull request by lightme16**
## What do these changes do?

- use python 3.7
- run tox in travis
- change deploy credentials

## Are there changes in behavior for the user?

No

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,29 +1,47 @@
+conditions: v1
+
+dist: trusty
```
Comment (webknjaz) [inline .travis.yml]: Use xenial instead. It supports Python 3.7 out of the box.

File: .travis.yml [hunk 2]
```diff
@@ -1,29 +1,47 @@
...
 python:
   - "3.5"
   - "3.6"
 install:
-  - pip install codecov
-  - pip install -r requirements.txt
+  - pip install -U setuptools
+  - pip install -U pip
+  - pip install -U wheel
+  - pip install -U tox
```
Comment (webknjaz) [inline .travis.yml]: This is probably the only thing needed.

File: .travis.yml [hunk 3]
```diff
@@ -1,29 +1,47 @@
...
+  TOXENV: py
+  ES_TAG: 6.0.0
+
+jobs:
+  fast_finish: true
+  include:
+  - &py-3_7-base
+    python: "3.7"
+    dist: xenial
+    sudo: required
```
Comment (webknjaz) [inline .travis.yml]: This is useless as Travis CI doesn't have docker containers anymore

File: .travis.yml [hunk 4]
```diff
@@ -1,29 +1,47 @@
...
+
+  - stage: Deploy to PYPI
+    if: tag IS present
+    <<: *py-3_7-base
+    install: skip
+    script: skip
+    after_success: []
+    deploy:
+      provider: pypi
+      user: andrew.svetlov
```
Comment (webknjaz) [inline .travis.yml]: Replace creds with https://github.com/aio-libs/aioelasticsearch/issues/103#issuecomment-485109278

File: .travis.yml (hunk 2 above)
Comment (lightme16) [reply .travis.yml]: done

**Comment by lightme16** [comment]
@webknjaz thanks, done, feel free to code review now

File: tox.ini [hunk 5]
```diff
@@ -1,20 +1,11 @@
...
-basepython:
-    py35: python3.5
-    py36: python3.6
-
-setenv =
-    debug: PYTHONASYNCIODEBUG=x
-    release: PYTHONASYNCIODEBUG=
-
 commands =
     flake8 --show-source aioelasticsearch
```
Comment (webknjaz) [inline tox.ini]: It's better to have separate envs for separate things

File: .travis.yml [hunk 6]
```diff
@@ -1,29 +1,38 @@
...
+    if: tag IS present
+    install: skip
+    script: skip
+    after_success: []
+    deploy:
+      provider: pypi
+      user: aio-libs-bot
+      password:
+        secure: "cKvNQXCPwKAZHB02Me462NOyd4za1r5TJYUgCvrAkJTSf6qHdlLnOvH1bZtDyfs2GEpvYlx8nNZJHkbVHSiz3UEkaiuLj6xfaJdVUpJ9qK5w4YQvrMeb4cD6amIvMttrgMoppGq4GSjdcggBIwNcmgk2g5PrpozheK2GWkA1B8rlsnud4JRPhenzchH5yvv1VBXokVFvv6SyS+EIL8DEKRCZQ/Bug3N5QVPXWQn52JINY0c6v0UgLJzc0F82VCJKYdKkVBXLFGakwHlufjcn5TB9myia5hBeObdyXlJGk+NiGrZGchluAfE+QWfdtTn9AAyUHdNV7jjQWQtfv5gj025Zo2IXNRvHkeIG6CiHyZ3YiXorA20kZgaHd95PG+WkJwbRUBJdCgrZ1r+saxrA8D7tT3aljEwc10jz/YcQdazQmFXqV2NP0PDAiqNS0zgIPPZQn2lo+KUBpiUA+h/Wp/Jv3XlmhyyvAeCfGVMDdBRhIq5SDHdaqVJ+UbGR79qKcA9/bKYimyZ83MRUBUsZPMh/VElLKImN3LqjCjsoEeFCeh5pZvEmrydNRv4aeE124nnYX0DY6Md6+NN6wjnBagg+Ws4cR0UzsQrO35L/JECEEklHQv+nGn4sTmMoHwf9IR0XeNwol3dnY/3HVuipgNcFN84Ir7F4gcUtshJu0Vo="
+      distributions: "sdist bdist_wheel"
```
Comment (webknjaz) [inline .travis.yml]:
```suggestion
      distributions: sdist bdist_wheel
```

File: tox.ini (hunk 5 above)
Comment (lightme16) [reply tox.ini]: fixed, returned it back

File: .travis.yml (hunk 6 above)
Comment (lightme16) [reply .travis.yml]: changed

**Comment by lightme16** [comment]
@webknjaz fixed

File: .travis.yml (hunk 6 above)
Comment (webknjaz) [reply .travis.yml]: instead of copy-pasting things you can actually commit suggestions from the GitHub UI

File: .travis.yml [hunk 7]
```diff
@@ -1,29 +1,38 @@
...
+    deploy:
+      provider: pypi
+      user: aio-libs-bot
+      password:
+        secure: "cKvNQXCPwKAZHB02Me462NOyd4za1r5TJYUgCvrAkJTSf6qHdlLnOvH1bZtDyfs2GEpvYlx8nNZJHkbVHSiz3UEkaiuLj6xfaJdVUpJ9qK5w4YQvrMeb4cD6amIvMttrgMoppGq4GSjdcggBIwNcmgk2g5PrpozheK2GWkA1B8rlsnud4JRPhenzchH5yvv1VBXokVFvv6SyS+EIL8DEKRCZQ/Bug3N5QVPXWQn52JINY0c6v0UgLJzc0F82VCJKYdKkVBXLFGakwHlufjcn5TB9myia5hBeObdyXlJGk+NiGrZGchluAfE+QWfdtTn9AAyUHdNV7jjQWQtfv5gj025Zo2IXNRvHkeIG6CiHyZ3YiXorA20kZgaHd95PG+WkJwbRUBJdCgrZ1r+saxrA8D7tT3aljEwc10jz/YcQdazQmFXqV2NP0PDAiqNS0zgIPPZQn2lo+KUBpiUA+h/Wp/Jv3XlmhyyvAeCfGVMDdBRhIq5SDHdaqVJ+UbGR79qKcA9/bKYimyZ83MRUBUsZPMh/VElLKImN3LqjCjsoEeFCeh5pZvEmrydNRv4aeE124nnYX0DY6Md6+NN6wjnBagg+Ws4cR0UzsQrO35L/JECEEklHQv+nGn4sTmMoHwf9IR0XeNwol3dnY/3HVuipgNcFN84Ir7F4gcUtshJu0Vo="
+      distributions: sdist bdist_wheel
+      on:
+        tags: true
+        all_branches: true
+        python: 3.7
```
Comment (webknjaz) [inline .travis.yml]: Float `3.7` will not match the string `"3.7"` from line 8 so this won't work.
```suggestion
        python: "3.7"
```

File: tox.ini [hunk 8]
```diff
@@ -25,4 +20,4 @@ commands =
     flake8 --show-source tests
```
Comment (webknjaz) [inline tox.ini]: linters must be in separate tox envs.

Comment (lightme16) [reply tox.ini]: done

File: tox.ini [hunk 9]
```diff
@@ -1,28 +1,32 @@
...
 
 [testenv]
 deps =
     -r{toxinidir}/requirements.txt
+commands = pytest tests {posargs}
 
-basepython:
-    py35: python3.5
-    py36: python3.6
+[testenv:flake8]
```
Comment (webknjaz) [inline tox.ini]: Now, include lint envs as separate jobs in Travis CI.

File: tox.ini [hunk 10]
```diff
@@ -1,28 +1,32 @@
...
+skip_install = True
+deps =
+    -r{toxinidir}/requirements.txt
+commands = flake8 --show-source aioelasticsearch tests setup.py
+
+[testenv:isort]
+skipsdist = True
+skip_install = True
+deps =
+    -r{toxinidir}/requirements.txt
```
Comment (webknjaz) [inline tox.ini]: I believe this env doesn't require the whole load of deps. just `isort` should be enough.

File: tox.ini [hunk 11]
```diff
@@ -1,28 +1,32 @@
...
+commands = pytest tests {posargs}
 
-basepython:
-    py35: python3.5
-    py36: python3.6
+[testenv:flake8]
+skipsdist = True
+skip_install = True
+deps =
+    -r{toxinidir}/requirements.txt
```
Comment (webknjaz) [inline tox.ini]: you should probably just use `flake8`

File: tox.ini (hunk 9 above)
Comment (lightme16) [reply tox.ini]: Not sure whether I have permissions to do this in Travis CI

File: tox.ini (hunk 11 above)
Comment (lightme16) [reply tox.ini]: Yes, changed "deps"

File: tox.ini (hunk 10 above)
Comment (lightme16) [reply tox.ini]: No, just isort is not enough, it needs to install dependencies in order to sort imports correctly.

**Comment by trsa** [comment]
@lightme16 any progress here?

**Comment by lightme16** [comment]
The work is finished. Waiting for @webknjaz review it

File: tox.ini (hunk 9 above)
Comment (webknjaz) [reply tox.ini]: It's about editing `.travis.yml`.

File: .travis.yml [hunk 12]
```diff
@@ -1,29 +1,38 @@
...
 install:
-  - pip install codecov
-  - pip install -r requirements.txt
+  - pip install -U tox
 script:
-  - pytest tests --es_tag 6.0.0
-  - PYTHONASYNCIODEBUG=x pytest tests --es_tag 6.0.0
+  - tox -- --es_tag=${ES_TAG}
 after_success:
+  - pip install codecov
```
Comment (webknjaz) [inline .travis.yml]: It'd be nice to put this whole thing into a separate toxenv and invoke it via `tox -e coverage`

File: tox.ini (hunk 9 above)
Comment (lightme16) [reply tox.ini]: oh, I see, added.

File: .travis.yml (hunk 12 above)
Comment (lightme16) [reply .travis.yml]: agree, added

**Comment by lightme16** [comment]
@webknjaz  Added some fixed according to your comments, feel free to review 👀

File: .travis.yml [hunk 13]
```diff
@@ -1,29 +1,37 @@
...
-  distributions: "sdist bdist_wheel"
-  on:
-    tags: true
-    all_branches: true
-    python: 3.5
+jobs:
+  fast_finish: true
+  include:
+  - stage: Deploy to PYPI
+    if: tag IS present
```
Comment (webknjaz) [inline .travis.yml]: Specify the runtime:
```suggestion
    if: tag IS present
    python: "3.7"
```

File: .travis.yml [hunk 14]
```diff
@@ -1,29 +1,37 @@
...
+    deploy:
+      provider: pypi
+      user: aio-libs-bot
+      password:
+        secure: "cKvNQXCPwKAZHB02Me462NOyd4za1r5TJYUgCvrAkJTSf6qHdlLnOvH1bZtDyfs2GEpvYlx8nNZJHkbVHSiz3UEkaiuLj6xfaJdVUpJ9qK5w4YQvrMeb4cD6amIvMttrgMoppGq4GSjdcggBIwNcmgk2g5PrpozheK2GWkA1B8rlsnud4JRPhenzchH5yvv1VBXokVFvv6SyS+EIL8DEKRCZQ/Bug3N5QVPXWQn52JINY0c6v0UgLJzc0F82VCJKYdKkVBXLFGakwHlufjcn5TB9myia5hBeObdyXlJGk+NiGrZGchluAfE+QWfdtTn9AAyUHdNV7jjQWQtfv5gj025Zo2IXNRvHkeIG6CiHyZ3YiXorA20kZgaHd95PG+WkJwbRUBJdCgrZ1r+saxrA8D7tT3aljEwc10jz/YcQdazQmFXqV2NP0PDAiqNS0zgIPPZQn2lo+KUBpiUA+h/Wp/Jv3XlmhyyvAeCfGVMDdBRhIq5SDHdaqVJ+UbGR79qKcA9/bKYimyZ83MRUBUsZPMh/VElLKImN3LqjCjsoEeFCeh5pZvEmrydNRv4aeE124nnYX0DY6Md6+NN6wjnBagg+Ws4cR0UzsQrO35L/JECEEklHQv+nGn4sTmMoHwf9IR0XeNwol3dnY/3HVuipgNcFN84Ir7F4gcUtshJu0Vo="
+      distributions: sdist bdist_wheel
+      on:
+        tags: true
+        all_branches: true
+        python: "3.7"
```
Comment (webknjaz) [inline .travis.yml]: This check is not needed
```suggestion
```

**Comment by webknjaz** [comment]
Almost there! Just two small edits + make this branch up-to-date with master..

**Comment by lightme16** [comment]
@webknjaz Done 🙂

**Pull request merged**